lib/engines/hcktest/hcktest.rb

```diff
@@ -143,29 +143,30 @@ def run_clients(scope, run_opts = {})
                                 this platform is incorrect'
     end
 
-    def configure_clients
+    def configure_and_synchronize_clients
       run_only = @project.options.test.manual && @project.options.test.driver_path.nil?
 
       @clients.each_value do |client|
         client.configure(run_only:)
       end
+
+      @clients.each_value(&:synchronize)
     end
 
     def configure_setup_and_synchronize
+      return configure_and_synchronize_clients unless @studio.tools.nil?
+
       @studio.configure(@platform['clients'])
-      configure_clients
-      @clients.each_value(&:synchronize)
+      configure_and_synchronize_clients
       @studio.keep_snapshot
       @clients.each_value(&:keep_snapshot)
     end
 
-    def run_and_configure_setup(scope)
+    def run_clients_and_configure_setup(scope, client_run_opts = {})
```

Comment:
Simpler:

```suggestion
    def run_clients_and_configure_setup(scope, **opts)
```

We already say `run_clients` so we don't need to say that in this method. Also let's use `**` trick.